modin/backends/pandas/query_compiler.py

```diff
@@ -1905,12 +1905,66 @@ def mode_builder(df):
         return self.__constructor__(new_modin_frame).dropna(axis=axis, how="all")
 
     def fillna(self, **kwargs):
+        squeeze_self = kwargs.pop("squeeze_self", False)
+        squeeze_value = kwargs.pop("squeeze_value", False)
         axis = kwargs.get("axis", 0)
         value = kwargs.get("value")
         method = kwargs.get("method", None)
         limit = kwargs.get("limit", None)
         full_axis = method is not None or limit is not None
-        if isinstance(value, dict):
+        if isinstance(value, BaseQueryCompiler):
+            kwargs.pop("value")
+            if squeeze_self:
+                # Self is a Series type object
+                if full_axis:
+                    value = value.to_pandas().squeeze(axis=1)
+
+                    def fillna_builder(series):
+                        # `limit` parameter works only on `Series` type, so we have to squeeze both objects to get
+                        # correct behavior.
+                        return series.squeeze(axis=1).fillna(value=value, **kwargs)
+
+                    new_modin_frame = self._modin_frame.apply_full_axis(
+                        0, fillna_builder
+                    )
+                else:
+
+                    def fillna_builder(series, value):
+                        # Both arguments for this function are 1-column `DataFrames` which denote `Series` type.
+                        # Because they are both of the same type, it is not necessary to convert either of them into
+                        # `Series` by squeezing since `fillna` works perfectly in the same way on 1-column `DataFrame`
+                        # objects (when `limit` parameter is absent) as it works on two `Series`.
+                        return series.fillna(value, **kwargs)
+
+                    new_modin_frame = self._modin_frame.binary_op(
+                        fillna_builder, value._modin_frame, join_type="left"
+                    )
+
+                return self.__constructor__(new_modin_frame)
+            else:
+                # Self is a DataFrame type object
+                if squeeze_value:
+                    # Value is Series type object
+                    value = value.to_pandas().squeeze(axis=1)
+
+                    def fillna(df):
+                        return df.fillna(value, **kwargs)
+
+                    # Continue to end of this function
+
+                else:
+                    # Value is a DataFrame type object
+                    def fillna_builder(df, r):
+                        # Behavior is different for `DataFrame` and `Series` type of `value` argument, so we have to squeeze
+                        # to make sure that `Series` object have a `Series` type.
+                        return df.fillna(r.squeeze(axis=1), **kwargs)
```

Comment:
do we need to do `.squeeze()` at this else-branch if `squeeze_value` is supposed to be False here?

Reply:
You are right. Comment and squeeze are obsolete leftovers when logic was different. Should be fixed in the next commit.